petastorm/tests/test_predicates.py

```diff
@@ -124,3 +131,51 @@ def test_predicate_on_single_column(synthetic_dataset):
         actual = dict(row._asdict())
         assert actual['id2'] < 2
     assert counter == len(synthetic_dataset.data)
+
+
+def test_predicate_on_partitioned_dataset():
+    """
+    Generates a partitioned dataset and ensures that readers evaluate the type of the partition
+    column according to the type given in the Unischema.
+    """
+    TestSchema = Unischema('TestSchema', [
+        UnischemaField('id', np.int32, (), ScalarCodec(IntegerType()), False),
+        UnischemaField('test_field', np.int32, (), ScalarCodec(IntegerType()), False),
+    ])
+
+    def test_row_generator(x):
+        """Returns a single entry in the generated dataset."""
+        return {'id': x,
+                'test_field': x*x}
+
+    rowgroup_size_mb = 256
+    dataset_url = "file:///tmp/partitioned_test_dataset"
+
+    spark = SparkSession.builder.config('spark.driver.memory', '2g').master('local[2]').getOrCreate()
+    sc = spark.sparkContext
+
+    rows_count = 10
+    with materialize_dataset(spark, dataset_url, TestSchema, rowgroup_size_mb):
+
+        rows_rdd = sc.parallelize(range(rows_count))\
+            .map(test_row_generator)\
+            .map(lambda x: dict_to_spark_row(TestSchema, x))
+
+        spark.createDataFrame(rows_rdd, TestSchema.as_spark_schema()) \
+            .coalesce(10) \
+            .write \
+            .partitionBy('id') \
+            .mode('overwrite') \
+            .parquet(dataset_url)
+
+    with make_reader(dataset_url, predicate=in_lambda(['id'], lambda x: x == 3)) as reader:
+        for sample in reader:
```

Comment:
One sample is enough to read. It will shorten the test run time.
(`next(reader)`)

Reply:
Fixed.